test/metabase/driver/sql/parameters/substitute_test.clj

```diff
@@ -856,15 +866,35 @@
                                 :target [:dimension [:template-tag "checkin_date"]]
                                 :value  "2015-07-01"}]))))))))
 
+(defmethod driver/database-supports? [::driver/driver ::e2e-parse-native-dates-test]
+  [driver _feature _database]
+  (contains? (sql-parameters-engines) driver))
+
+;;; TODO -- not clear why SQLite is getting skipped here, there was no comment explaining why in the code I migrated.
+(defmethod driver/database-supports? [:sqlite ::e2e-parse-native-dates-test]
+  [_driver _feature _database]
+  false)
+
+(defmulti e2e-parse-native-dates-test-sql
+  {:arglists '([driver])}
+  tx/dispatch-on-driver-with-test-extensions
+  :hierarchy #'driver/hierarchy)
+
+(defmethod e2e-parse-native-dates-test-sql ::driver/driver
+  [_driver]
+  "SELECT cast({{date}} as date)")
+
+(defmethod e2e-parse-native-dates-test-sql :oracle
+  [_driver]
+  "SELECT cast({{date}} as date) from dual")
```

Comment:
Yeah they probably should live somewhere driver test namespaces, for example if we removed the Oracle driver then it would be pretty nice if we didn't have to go update a bunch of unrelated tests. But I didn't want to touch too many things in this PR since it's already so big. More cleanup can be done in a follow up